### PR TITLE
Making sure outErrorPos is initialized if there is an error. CBL-2217

### PR DESCRIPTION
**Pull request by jianminzhao**

### CODE REVIEW

File: C/c4Query.cc [hunk 1]
```diff
@@ -60,10 +60,13 @@ Retained<C4Query> C4Query::newQuery(C4Collection *coll, C4QueryLanguage language
     } catch (Query::parseError &x) {
         if (outErrorPos) {
             *outErrorPos = x.errorPosition;
-            return nullptr;
-        } else {
-            throw;
         }
+        throw;
+    } catch (...) {
+        if (outErrorPos != nullptr) {
```
Comment (borrrden) [inline C/c4Query.cc:66]: Would prefer this to match the other clause as `if (outErrorPos)`

Comment (jianminzhao) [reply C/c4Query.cc]: Changed as suggested.

File: C/c4Query.cc [hunk 2]
```diff
@@ -60,10 +60,13 @@ Retained<C4Query> C4Query::newQuery(C4Collection *coll, C4QueryLanguage language
     } catch (Query::parseError &x) {
         if (outErrorPos) {
             *outErrorPos = x.errorPosition;
-            return nullptr;
```
Comment (borrrden) [inline C/c4Query.cc:63]: Was this removal on purpose?

Comment (jianminzhao) [reply C/c4Query.cc:63]: yes, because I don't think we should change the logic here merely because being provided outErrorPos. The exception may contain other useful info, for example, why the syntax error, that the platform may deicide whatever to do with it.

**Review by borrrden — APPROVED** [review]
